### PR TITLE
Libselinux

### PR DESCRIPTION
**Pull request by anjandev**
libselinux.so.1 is required for matlab according to: https://wiki.archlinux.org/index.php/MATLAB#Blackscreen_in_help_browser_and_livescripts

As such, Im trying to port this aur package: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=libselinux

Unfortunately Im having trouble with including `libselinux.so.1` in my package as it says xbps-src is ignoring this file:

```
=> libselinux-3.0_1: running pre-pkg hook: 04-generate-runtime-deps ...
   SONAME: libselinux.so.1 <-> libselinux-3.0_1 (ignored)
```

shlibs are hard. Please let me know what I am doing wrong. Thank you.

### CODE REVIEW

**Comment by abenson** [comment]
That's just saying it doesn't need to find the shlib in another package since it is already in this package. 

For matlab to detect it, you'll need to add it to `common/shlibs`.

**Comment by anjandev** [comment]
@abenson okay. Thank you. Do you know why matlab doesnt detect it even though I added it to `common/shlibs`?

**Comment by abenson** [comment]
No, I don't. Is there a template for matlab?

**Comment by anjandev** [comment]
@abenson matlab is a proprietary application. If I make a matlab meta package and add libselinux as a dependancy, will it copy the required shlib to /lib?

**Comment by abenson** [comment]
I don't understand what you mean about meta package. If you're installing matlab directly, make sure the libselinux package is installed. If its still says its missing, then make sure you have the right version of libselinux.

Without seeing how its detecting it or not detecting it, I'm not sure how to help

**Comment by anjandev** [comment]
@abenson I just copied `libselinux.so.1` generated in my chroot to my host system's `/lib` directory and matlab works now.

matlab detects libselinux if `libselinux.so.1` is in `/lib`. However, when I install the packages generated by these templates, there is no `libselinux.so.1` in `/lib`.

Any idea on fixing my template so that `libselinux.so.1` is placed in `/lib` on the system I install these packages?

File: srcpkgs/libselinux/template [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+depends="libsepol pcre"
+short_desc="SELinux library and simple utilities"
+maintainer="Anjandev Momi <anjan@momi.ca>"
+license="GPL-2.0-only"
+homepage="https://www.nsa.gov/what-we-do/research/selinux/"
+distfiles="https://github.com/SELinuxProject/selinux/releases/download/20191204/${pkgname}-${version}.tar.gz"
+checksum=2ea2b30f671dae9d6b1391cbe8fb2ce5d36a3ee4fb1cd3c32f0d933c31b82433
+
+post_install() {
+	rm -rf ${DESTDIR}/lib/*
```
Comment (sgn) [inline srcpkgs/libselinux/template:18]: Use this instead:
```sh
mv $DESTDIR/lib/* $DESTDIR/usr/lib
unlink /usr/lib/libselinux.so
ln -s libselinux.so.1 $DESTDIR/usr/lib/libselinux.so
```
And your problem will be solved.
Same comment applied to `libsepol`

Comment (not-chicken) [reply srcpkgs/libselinux/template:18]: I haven't tested it but maybe passing `SHLIBDIR=/usr/lib` to `make_install_args` will do the trick as well.

Comment (sgn) [reply srcpkgs/libselinux/template:18]: @not-chicken If it work, it's better solution, I only look into this template :/

Comment (not-chicken) [reply srcpkgs/libselinux/template:18]: Seems to work, upstream had it set to `/lib` 
```
$ xbps-query -f libselinux
...
/usr/lib/libselinux.so.1
...

Comment (anjandev) [reply srcpkgs/libselinux/template]: Thank you. I implemented @not-chicken's  changes in the latest patch. It works great!

File: common/shlibs [hunk 2]
```diff
@@ -3889,3 +3889,4 @@ libjcat.so.1 libjcat-0.1.2_1
 libmatio.so.9 matio-1.5.17_2
 libvips.so.42 libvips-8.9.2_1
 libvips-cpp.so.42 libvips-8.9.2_1
+libsepol.so.1 libsepol-3.0_1
```
Comment (sgn) [inline common/shlibs:3892]: This change belongs to `libsepol`. The next line (`libselinux.so.1`) belongs to this change.

File: srcpkgs/libselinux/template [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Template file for 'libselinux'
+pkgname=libselinux
```
Comment (sgn) [inline srcpkgs/libselinux/template:2]: Please split `libselinux-devel` and `libsepol-devel`

File: srcpkgs/libsepol/template [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'libsepol'
+pkgname=libsepol
+version=3.0
+revision=1
+build_style=gnu-makefile
+make_install_args=SHLIBDIR=/usr/lib
+hostmakedepends="flex"
+depends="glibc"
```
Comment (sgn) [inline srcpkgs/libsepol/template:8]: Is it really depends on `glibc`?
This will block `*-musl`

Comment (sgn) [reply srcpkgs/libsepol/template:8]: I pushed without this, feel free to correct me if I messed up.

**Comment by anjandev** [comment]
> I pushed without this, feel free to correct me if I messed up.

Okay thank you. I am testing right now and it seems fine.

**Pull request closed without merging**